vernac/search.ml

```diff
@@ -64,51 +63,36 @@ let iter_constructors indsp u fn env nconstr =
-
-let handle h (Libobject.Dyn.Dyn (tag, o)) = match DynHandle.find tag h with
-| f -> f o
-| exception Not_found -> ()
-
 (* General search over declarations *)
 let generic_search env (fn : GlobRef.t -> Decls.logical_kind option -> env -> constr -> unit) =
-  List.iter (fun d -> fn (GlobRef.VarRef (NamedDecl.get_id d)) None env (NamedDecl.get_type d))
-    (Environ.named_context env);
-  let iter_obj prefix lobj = match lobj with
-    | AtomicObject o ->
-      let handler =
-        DynHandle.add Declare.Internal.Constant.tag begin fun (id,obj) ->
-          let kn = KerName.make prefix.Nametab.obj_mp (Label.of_id id) in
-          let cst = Global.constant_of_delta_kn kn in
-          let gr = GlobRef.ConstRef cst in
-          let (typ, _) = Typeops.type_of_global_in_context (Global.env ()) gr in
-          let kind = Declare.Internal.Constant.kind obj in
-          fn gr (Some kind) env typ
-        end @@
-        DynHandle.add DeclareInd.Internal.objInductive begin fun (id,_) ->
-          let kn = KerName.make prefix.Nametab.obj_mp (Label.of_id id) in
-          let mind = Global.mind_of_delta_kn kn in
-          let mib = Global.lookup_mind mind in
-          let iter_packet i mip =
-            let ind = (mind, i) in
-            let u = Univ.make_abstract_instance (Declareops.inductive_polymorphic_context mib) in
-            let i = (ind, u) in
-            let typ = Inductiveops.type_of_inductive env i in
-            let () = fn (GlobRef.IndRef ind) None env typ in
-            let len = Array.length mip.mind_user_lc in
-            iter_constructors ind u fn env len
-          in
-          Array.iteri iter_packet mib.mind_packets
-        end @@
-        DynHandle.empty
-      in
-      handle handler o
-    | _ -> ()
+  let iter_var d = fn (GlobRef.VarRef (NamedDecl.get_id d)) None env (NamedDecl.get_type d) in
+  let iter_constant cst _ =
+    let gr = GlobRef.ConstRef cst in
+    let (typ, _) = Typeops.type_of_global_in_context (Global.env ()) gr in
+    let kind = Dumpglob.constant_kind cst in
+    fn gr (Some kind) env typ
+  in
+  let iter_inductive mind (mib,_) =
+    let iter_packet i mip =
+      let ind = (mind, i) in
+      let u = Univ.make_abstract_instance (Declareops.inductive_polymorphic_context mib) in
+      let i = (ind, u) in
+      let typ = Inductiveops.type_of_inductive env i in
+      let () = fn (GlobRef.IndRef ind) None env typ in
+      let len = Array.length mip.mind_user_lc in
+      iter_constructors ind u fn env len
+    in
+    Array.iteri iter_packet mib.mind_packets
   in
-  try Declaremods.iter_all_segments iter_obj
-  with Not_found -> ()
+  List.iter iter_var (Environ.named_context env);
+  let globals = Environ.Globals.view env.env_globals in
+  try
+    Cmap_env.iter iter_constant globals.constants;
+    Mindmap_env.iter iter_inductive globals.inductives
+  with Not_found ->
+    (* Not sure what is supposed to raise Not_found, TODO try removing
```

Comment:
Do you want to resolve this TODO as part of this PR, since the calls in the `try` block have changed?